src/vs/editor/common/model/bracketPairColorizer/immutableSet.ts

```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
Maybe move utilities that are editor agnostic into `base` with some tests`? I.e. we have `map.ts`.